Bio/SeqAnalysis/SeqAnalyzer.py

```diff
@@ -0,0 +1,168 @@
+        2. pi - calculated isoelectric point for protein sequences in a list
+        3. amino_acids - counted standard aa in protein sequences, in a list of dicts
+        4. aromacity - calculated aromacity for protein sequences according to Lobry 1994 method, in a list
+        5. amino_acids_percent - calculated percantage based aa content in protein sequences, in a list
+        6. instability - calculated instability indexes for protein sequences according to Guruprasad et al 1990, in a list
+        7. flexibility - calculated flexibility of protein sequences according to Vihinen, 1994, in a list
+        8. secondary_structure_fraction - calculated fraction of helix, turn and sheet for protein sequences, in a list
+        of tuples with 3 elements (helix, turn, sheet) fractions accordingly
+        9. seq_gravy - calculated gravy of protein sequences according to Kyte and Doolittle, in a list
+        10.mol_ext_coefficient - calculated molar extinction coefficients for protein sequences, in a list
+        """
+        l_objects = [ProteinAnalysis(_[1]) for _ in self.tuples]
+        molecular_weight = [_.molecular_weight() for _ in l_objects]
+        pi = [_.isoelectric_point() for _ in l_objects]
+        amino_acids = [_.count_amino_acids() for _ in l_objects]
+        aromacity = [_.aromaticity() for _ in l_objects]
+        amino_acids_percent = [_.get_amino_acids_percent() for _ in l_objects]
+        instability = [_.instability_index() for _ in l_objects]
+        flexibility = [sum(__) / len(__) for __ in [_.flexibility() for _ in l_objects]]
+        secondary_structure_fraction = [_.secondary_structure_fraction() for _ in l_objects]
+        seq_gravy = [_.gravy() for _ in l_objects]
+        mol_ext_coefficient = [_.molar_extinction_coefficient()[0] for _ in l_objects]
+        return molecular_weight, pi, amino_acids, aromacity, amino_acids_percent, instability, flexibility, \
+               secondary_structure_fraction, seq_gravy, mol_ext_coefficient
+
+    def n_and_c_term(self) -> tuple:
+        """
+        Counts occurrences of given aa on 'C' and 'N' terminus of protein
+
+        :return: returns a list of two dicts, where list[0] is a dictionary with information about occurrences
+        of aa on 'N' terminus, and list[1] about 'C' terminus respectively
+        """
+        list_c_term = [_[-1] for _ in self.dict.values()]
+        dict_c_term = {i: list_c_term.count(i) / len(set(list_c_term)) for i in set(list_c_term)}
+
+        list_n_term = [_[0] for _ in self.dict.values()]
+        dict_n_term = {i: list_n_term.count(i) / len(set(list_n_term)) for i in set(list_n_term)}
+        return dict_n_term, dict_c_term
+
+    def __init__(self, records):
+        self.dict = records  # Analyzer.makedict(self, records)  # creates a dict with seq ids and seqeances itself
+        self.tuples = [(id, seq) for id, seq in self.dict.items()]
```

Comment:
Why, just why? You have input in dictionary format, why creating tuples?

Reply:
to make sure i operate on ordered ID-aa sequence data structure while calculating all values

Reply:
But you can use dict directly and don't rely on lists. If you want to contain info about sequence use dict inside dict instead of 4 different lists:
```
seqs = {
   'id': {
      'seq': 'AAABBBCCC',
      'len': 9,
      ...
   }
}
```

Then just export data in proper format:
```
sequences_lengths = [aa_analysis_data['length'] for aa_analysis_data in aa.values()]
```